api/envoy/config/trace/v2/trace.proto

```diff
@@ -100,6 +103,43 @@ message DatadogConfig {
   string service_name = 2 [(validate.rules).string.min_bytes = 1];
 }
 
+// Configuration for the OpenCensus tracer.
+message OpenCensusConfig {
+  // Configures tracing, e.g. the sampler, max number of annotations, etc.
+  opencensus.proto.trace.v1.TraceConfig trace_config = 1;
+
+  // Enables the stdout exporter if set to true. This is intended for debugging
+  // purposes.
+  bool stdout_exporter_enabled = 2;
+
+  // Enables the Stackdriver exporter if set to true. The project_id must also
+  // be set.
+  bool stackdriver_exporter_enabled = 3;
+
+  // The Cloud project_id to use for Stackdriver tracing.
+  string stackdriver_project_id = 4;
+
+  // Enables the Zipkin exporter if set to true. The url and service name must
+  // also be set.
+  bool zipkin_exporter_enabled = 5;
+
+  // The URL to Zipkin, e.g. "http://127.0.0.1:9411/api/v2/spans"
+  string zipkin_url = 6;
+
+  // The Zipkin service name.
+  string zipkin_service_name = 7;
```

Comment:
Not mutually exclusive.

Although maybe odd, this can be used for dual export. Mostly I use it to enable the stdout/debugging exporter at the same time as a real backend.

Reply:
@g-easy Sorry for the late question, but only just having time to look at trying the OpenCensus tracer - wondering why the service name is being supplied here, rather than taken from the `LocalInfo.clusterName()` as with the zipkin tracer?

Reason being that it would be easier to use from Istio, where the tracer config is shared across all proxies, and the individual service name is supplied as a command line option.

Reply:
I wasn't aware of `LocalInfo.clusterName` at the time - sorry about that.

Reply:
How about this: https://github.com/envoyproxy/envoy/pull/7354

Reply:
Looks great - thanks for the quick response.